### PR TITLE
dht: token: make some cosmetic changes

### PR DESCRIPTION
**Pull request by michoecho**
This is a set of a few cosmetic changes in dht/token. Mostly some comments and a simplification of `midpoint()`.

### CODE REVIEW

**Review by psarna — APPROVED** [review]

File: dht/token.cc [hunk 1]
```diff
@@ -79,35 +79,10 @@ sstring token::to_sstring() const {
...
 token token::midpoint(const token& t1, const token& t2) {
-    auto l1 = long_token(t1);
-    auto l2 = long_token(t2);
-    int64_t mid;
-    if (l1 <= l2) {
-        // To find the midpoint, we cannot use the trivial formula (l1+l2)/2
-        // because the addition can overflow the integer. To avoid this
-        // overflow, we first notice that the above formula is equivalent to
-        // l1 + (l2-l1)/2. Now, "l2-l1" can still overflow a signed integer
-        // (e.g., think of a very positive l2 and very negative l1), but
```
Comment (haaawk) [inline dht/token.cc:99]: What about this overflow of l2 - l1 ?

Comment (haaawk) [reply dht/token.cc:99]: I guess l1 can't be negative so we're save.

**Review by haaawk — APPROVED** [review]
LGTM

File: dht/token.cc (hunk 1 above)
Comment (michoecho) [reply dht/token.cc:99]: The new code is the same as the true branch of the old code. We still avoid overflow problems by doing the arithmetic in unsigned integers. I simply noticed that the formula for the true branch also works naturally for the false branch.

File: dht/token.hh [hunk 2]
```diff
@@ -77,8 +76,7 @@ public:
             if (b.size() != sizeof(_data)) {
                 throw std::runtime_error(fmt::format("Wrong token bytes size: expected {} but got {}", sizeof(_data), b.size()));
             }
-            std::copy_n(b.begin(), sizeof(_data), reinterpret_cast<int8_t *>(&_data));
-            _data = net::ntoh(_data);
+            _data = net::ntoh(read_unaligned<int64_t>(b.begin()));
```
Comment (avikivity) [inline dht/token.hh:87]: Could have been further simplified to read_be(). But fine as is.

File: dht/token.cc [hunk 3]
```diff
@@ -79,35 +79,10 @@ sstring token::to_sstring() const {
...
-        // When l2 < l1, we need to switch l1 and and l2 in the above
-        // formula, because now l1 - l2 is positive.
-        // Additionally, we consider this case is a "wrap around", so we need
-        // to behave as if l2 + 2^64 was meant instead of l2, i.e., add 2^63
-        // to the average.
-        mid = l2 + positive_subtract(l1, l2)/2 + 0x8000'0000'0000'0000;
-    }
+    uint64_t l1 = long_token(t1);
+    uint64_t l2 = long_token(t2);
+    int64_t mid = l1 + (l2 - l1)/2;
```
Comment (avikivity) [inline dht/token.cc:85]: Let's try two examples:

1. l1=1, l2=-1

then mid = 1 + (0xffff'ffff'ffff'ffff - 1) / 2 = 1 + 0x7fff'ffff'ffff'ffff = 0x8000'0000'0000'0000, matches the old code.

2. l1=0x7fff'ffff'fffff'ffff, l2=-0x7ffff'ffff'ffff'ffff

then mid = 0x7fff'ffff'fffff'ffff +  (0x8000'0000'0000'0001 - 0x7fff'ffff'fffff'ffff) / 2 = 0x8000'0000'0000'0000, again matches the old code.

Comment (avikivity) [reply dht/token.cc:85]: btw, I think the only remaining user of midpoint() is in token_test.

**Pull request closed without merging**